pkg/git/git_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package git_test
+
+import (
+	"context"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	crc "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	build "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/shipwright-io/build/pkg/config"
+	buildController "github.com/shipwright-io/build/pkg/controller/build"
+	"github.com/shipwright-io/build/pkg/controller/fakes"
+	"github.com/shipwright-io/build/pkg/ctxlog"
+	"github.com/shipwright-io/build/test"
+)
+
+var _ = Describe("Git", func() {
+	var (
+		sourceURL                    string
+		fakeSourceURL                string
+		registrySecret               string
+		buildName                    string
+		namespace, buildStrategyName string
+		ctl                          test.Catalog
+		manager                      *fakes.FakeManager
+		request                      reconcile.Request
+		client                       *fakes.FakeClient
+		statusWriter                 *fakes.FakeStatusWriter
+		buildSample                  *build.Build
+		reconciler                   reconcile.Reconciler
+	)
+	BeforeEach(func() {
```

Comment:
This is not a unit test case. Imo the unit test of the git package should directly test the public members of the git package. Those are `IdentifyGitURLType` and `ValidateGitURLExists`. I do not see a reason to create any Kubernetes object here.

Reply:
@SaschaSchwarze0 
> I do not see a reason to create any Kubernetes object here.
>
Sorry, I don't understand what's your mean? Could you pls explain it more clearly? AFAIK, unit test needs to prepare some fake resources, why don't need to create kubernetes objects? I think we need a Build object first, also some fake buildStrategy and secret.

Reply:
@xiujuan95 my point is that the unit test should test the unit that it is supposed to test. The unit imo here is the `git` package that you introduced. It has the two functions that I mentioned above. In your unit test, you can invoke these two functions directly without a need for a build and buildrun.

Reply:
@SaschaSchwarze0 Yes, exactly. I change it, thanks!